astropy/io/ascii/core.py

```diff
@@ -671,7 +671,7 @@ class BaseData(object):
     fill_include_names = None
     fill_exclude_names = None
     # Currently, the default matches the numpy default for masked values.
```

Comment:
If you change the default, this comment line has to go. Note however, that this also states why we chose "--" as default in the first place - it's consistent with numpy. Maybe astropy is mature enough that we don't need that any longer, if we want the "least surprising" behavior, doing what numpy does is not a bad choice.